File: Patterns/8. Inverted Pyramid Patterns/71. Inverted Pyramid Stars/ips.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=n; i++){
        // print spaces
        for(int j=1; j<=i-1; j++){
            cout<<"  ";
        } 
        // print stars
        for(int j=1; j<=2*(n-i+1)-1; j++){
            cout<<"* ";
        }
        cout<<'\n';
    }
}

/*
Output

* * * * * * * * * 
  * * * * * * * 
    * * * * * 
      * * * 
        * 

Another Method (Matrix form) using if-else

    int p;
    cout<<"Enter a Number : ";
    cin>>p;

    for(int i=1; i<=p; i++){
        for(int j=1; j<=2*p-1; j++){
            if(j>=i && j<=2*n-1){
                cout<<"* ";
            }else{
                cout<<"  ";
            }
        }
        cout<<'\n';
    }

*/

File: Patterns/8. Inverted Pyramid Patterns/71. Inverted Pyramid Stars/ips.py
n = int(input("Enter a Number : "))

for i in range(1,n+1):
    # print spaces
    for j in range(1,i):
        print(end='  ')
    # print stars
    for j in range(1,2*(n-i+1)):
        print("*", end=' ')
    print()

'''
Output:

* * * * * * * * * 
  * * * * * * * 
    * * * * * 
      * * * 
        * 

Another Method using if-else:

p = int(input("Enter a Number : "))

for i in range(1,p+1):
    for j in range(1,2*p):
        if(i <= j <= 2*p-i):
            print("*",end=" ")
        else:
            print(end='  ')
    print()

'''